### ISSUE TITLE
Editor shortcuts using option break after changing Keyboard Preferences

### ISSUE THREAD

**Issue by infovore** [benfry/processing4#404]
## Description

On Mac OS (tested on: 10.15.x), option-based editor motions (option-cursor left/right for "jump by word"; option-backspace for "delete word") do not work at all in the IDE. Command-based motions (ie, beginning of line / end of line) work just fine.

This is highly frustrating from a muscle memory point of view, and is a regression from 3.x.

See also issue #140.

## Expected Behavior

Option-left and option-right should jump cursor by word in the IDE; option-backspace should delete word to left of cursor.

## Current Behavior

Option-left, option-right, option-backspace do nothing (they are swallowed).

## Steps to Reproduce

1. Open the editor on a mac.
2. Write some code.
3. Try to move around it using option-left, option-right, or delete word to the left with option-backspace.

## Your Environment

Processing 4.0b5, Mac OS 10.15.7

**Comment by benfry** [comment]
A third of my time developing this is on 10.15, and I've never been able to reproduce this. So is there anything else you have installed that could be causing trouble? Something that handles key shortcuts? Does it start working if you quit and restart Processing? (Or if not, reboot your machine?) Anything else I can go on, like whether you're on a non-US keyboard or version of macOS?

**Comment by infovore** [comment]
Hi, Ben - 

thanks for the response. A bit more detective work at this end and I think I've got a diagnosis:

* UK keyboard layout, using an external keyboard. I've tried switching to a US layout, and this makes no difference.
* this only seems to affect option-left, option-right, option-delete motions; holding down _shift_ for "select by word" works correctly...
* ...as long as I push option _before_ shift. Shift-then-option fails.
* oh! I am using a MS keyboard, designed for an MS Windows layout, and as such I have swapped "option" and "command" in my System Preferences. **Reverting them fixes the issue** - that is to say, if I leave Command and Option on their default mappings (`Win` as command, `alt` as option), then the Processing IDE functions just fine.

So: I'm retitling this issue to be more accurate, now I know what's up. This feels like something has changed in the way the external keyboard inside the editor is handled, possibly just down to Java changes; P3.5.4 handles non-default key layouts appropriately.

**Comment by benfry** [comment]
Ouch, ok… so this sounds like something outside our control, i.e. it may just be a bug in Java 17. The next release (beta 6) will use 17.0.2 which may even fix the issue. 

Thanks for looking into it further! I'll keep an eye out.

**Comment by infovore** [comment]
Yep, feel like it might just be an issue inside Swing or something. I will keep an eye out for beta 6 and run this test again then!
